File: lib/layer2_core/light/e_on_tertiary_fixed_variant_light.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/light/system.dart';
import '../../layer1_raw/light/standard.dart';
import '../../layer1_raw/light/blue.dart';
import '../../layer1_raw/light/green.dart';
import '../../layer1_raw/light/red.dart';
import '../../layer1_raw/light/nothing.dart';

class EOnTertiaryFixedVariantLight {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyl40;
      case 'standard': return cstl40;
      case 'blue': return cbl40;
      case 'green': return cgl40;
      case 'red': return crl40;
      case 'nothing': return cnol40;
      default: return cstl40;
    }
  }
}

File: android/src/main/kotlin/com/example/design_engine/layer2_core/light/EOnTertiaryFixedVariantLight.kt
package com.example.design_engine.layer2_core.light
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object EOnTertiaryFixedVariantLight {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyl40
            AppTheme.STANDARD -> R.color.cstl40
            AppTheme.BLUE -> R.color.cbl40
            AppTheme.GREEN -> R.color.cgl40
            AppTheme.RED -> R.color.crl40
            AppTheme.NOTHING -> R.color.cnol40
        }
    }
}
